### DISCUSSION TITLE
How to save data from custom interface to a database

### DISCUSSION THREAD

**Discussion by SNPR** [directus/directus#3538] [Archive]
Hello! I created a custom interface for the string type  (in my case - regular input):

### interface.vue
```
<template>
  <div>
    <div>My Custom Interface</div>
    <input v-model="value" />
  </div>
</template>

<script>
  export default {
    props: {
      value: String,
      type: String
    },
    created() {
      console.log(this)
    }
  };
</script>
```

### index.js
```
import InterfaceComponent from "./interface.vue";

export default {
  id: "custom",
  name: "Custom",
  description: "This is my custom interface!",
  icon: "box",
  component: InterfaceComponent,
  types: ["string"],
};
```

And when I type something to this input I can't save the result to the database (save button is disabled):
![image](https://user-images.githubusercontent.com/22043414/103266706-06254e00-49d2-11eb-97db-bcf4b8d93dc3.png)

What should I do in order to have a possibility to save custom interface value to the database?

**Comment by rijkvanzanten** [comment]
You emit a value in the `input` event, so for example `this.$emit('input', 'text')`

**Reply by SNPR** [reply]
Thanks, it works!
I did the following:
` <input @input="$emit('input', $event.target.value)" v-model="value" />`

**Reply by benhaynes** [reply]
Do we need to update the Docs or some example code somewhere?

**Reply by SNPR** [reply]
> Do we need to update the Docs or some example code somewhere?

Yes, of course! Because without creating this discussion it was impossible for me to save value from custom interface to the database.

**Reply by benhaynes** [reply]
We don't have a ticket for that, @rijkvanzanten ... should I add a ticket or is this an easy paste somewhere?

**Reply by rijkvanzanten** [reply]
Ticket is probably best 👍🏻
